tools/generator/templates/includes/Makefile-language.jinja2

```diff
@@ -22,7 +22,7 @@ to_clean = $(objs)
 
 {% block build_library %}
 champion.so: $(objs)
-	$(CXX) -o $@ $^ -shared $(LDFLAGS)
+	$(CXX) -o $@ -Wl,--whole-archive $^ -Wl,--no-whole-archive -shared $(LDFLAGS)
```

Comment:
Could you add a comment to explain this? Is that really something we want for all the languages as opposed to only for Rust?